src/core/border_router/routing_manager.cpp

```diff
@@ -2200,6 +2200,11 @@ void RoutingManager::LocalOnLinkPrefix::HandleExtPanIdChange(void)
 
     case kAdvertising:
     case kDeprecating:
+        if (mOldPrefix.GetLength() != 0)
+        {
+            Unpublish(mOldPrefix);
+        }
```

Comment:
Maybe we need this code even in `kIdle`?

I feel `mState` is just about managing `mPrefix`. It has nothing to do with `mOldPrefix`?

Reply:
Yes. The `mOldPrefix` and `mState` are somewhat independent.

But I don't think we need this code under `kIdle` since we are not changing the `mOldPrefix` and it can stay as it was before (we keep deprecating it).

The change here in this PR is that we make sure to unpublish the `mOldPrefix` before we  change the `mOldPrefix` variable itself. Basically since we remember only one old entry, if we are still deprecating the previous one, we clean up before changing the variable (to remember the new old prefix).

Reply:
Make sense to me.